crates/circuit/src/dag_circuit.rs

```diff
@@ -5368,19 +5345,21 @@ impl DAGCircuit {
         Ok(child == output_node)
     }
 
-    fn may_have_additional_wires(&self, py: Python, instr: &PackedInstruction) -> bool {
+    fn may_have_additional_wires(&self, instr: &PackedInstruction) -> bool {
         let OperationRef::Instruction(inst) = instr.op.view() else {
```

Comment:
Nice, this means we don't even acquire the GIL unless we've got a Python-based instruction 🙂.